plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php

```diff
@@ -135,22 +135,28 @@ public function track_product_updated_client_side( $post ) {
 			if ( $( 'h1.wp-heading-inline' ).text().trim() === '" . __( 'Edit product', 'woocommerce' ) . "') {
 				var initialStockValue = $( '#_stock' ).val();
 				var hasRecordedEvent = false;
+				var isBlockEditor = false;
 
-				$( '#publish' ).on( 'click', function() {
+				function sendTracks() {
 					if ( hasRecordedEvent ) {
```

Comment:
This logic will also have to change for the block editor. The block editor doesn't refresh the page when you click `update`, meaning that with this logic, it will only get triggered on the first save but not there after.
But maybe that is what this is trying to do anyhow? I am not a 100% sure what the benefit of this was given the browser does a full refresh anyhow, so maybe it can just be removed?

Reply:
Split the logic 👌